### PR TITLE
initial transportation model

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/models/transport.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+                    ('from:port', ('transport:air:port', {}), {
+                        'doc': 'The origin airport of this flight.'}),
+
+                    ('stops', ('array', {'type': 'transport:air:port'}), {
+                        'doc': 'An ordered list of aiport codes for stops which occured during this flight.'}),
+                )),
+                ('transport:air:telem', {}, (
+                    ('flight', ('transport:air:flight', {}), {
+                        'doc': 'The flight being reported meastured.'}),
```
Comment (rakuy0) [inline synapse/models/transport.py:106]: s/meastured/measured/g

File: synapse/models/transport.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                    ('tailnum', ('transport:air:tailnum', {}), {
+                        'doc': 'The tail/registration number at the time the plane flew this flight.'}),
+                    ('to:port', ('transport:air:port', {}), {
+                        'doc': 'The destination airport of this flight.'}),
+                    ('from:port', ('transport:air:port', {}), {
+                        'doc': 'The origin airport of this flight.'}),
+                    ('stops', ('array', {'type': 'transport:air:port'}), {
+                        'doc': 'An ordered list of aiport codes for stops which occured during this flight.'}),
+                    ('cancelled', ('bool', {}), {
+                        'doc': 'Set to true for canclled flights.'}),
```
Comment (whippit) [inline synapse/models/transport.py:98]: spelling

File: synapse/models/geospace.py [hunk 3]
```diff
@@ -187,16 +188,33 @@ def _normPyStr(self, text):
             raise s_exc.BadTypeValu(valu=text, name=self.name,
                                     mesg='invalid/unknown dist unit: %s' % (unit,))
 
-        return int(valu * mult), {}
+        norm = int(valu * mult) + self.baseoff
+        if norm < 0:
+            mesg = 'A geo:dist may not be negative.'
+            raise s_exc.BadTypeValu(mesg=mesg)
```
Comment (vEpiphyte) [inline synapse/models/geospace.py:194]: name=self.name

File: synapse/models/gov/intl.py [hunk 4]
```diff
@@ -8,6 +8,9 @@ def getModelDefs(self):
                 ('iso:oid', ('str', {'regex': '^([0-2])((\.0)|(\.[1-9][0-9]*))*$'}), {
                     'doc': 'An ISO Object Identifier string.'}),
 
+                ('iso:3166:cc', ('str', {'lower': True, 'regex': '^[a-z]{2}$'}), {
```
Comment (vEpiphyte) [inline synapse/models/gov/intl.py:11]: How is this different from ``pol:iso2`` ?

File: synapse/lib/task.py [hunk 5]
```diff
@@ -96,6 +96,9 @@ def pack(self):
 
         return pask
 
+    def cancel(self):
+        return self.task.cancel()
```
Comment (vEpiphyte) [inline synapse/lib/task.py:100]: This API is  appears unused - what is it here for?

Comment (invisig0th) [reply synapse/lib/task.py:100]: This is a fix for the fact that the fini() routine in the agenda subsystem calls cancel() on the synapse Task objects, which have no cancel() method.

File: synapse/models/transport.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                ('transport:sea:vessel', {}, (
+                    ('imo', ('transport:sea:imo', {}), {
+                        'doc': 'The International Maritime Organization number for the vessel.'}),
+                    ('name', ('str', {'lower': True, 'onespace': True, 'strip': True}), {
+                        'doc': 'The name of the vessel'}),
+                    ('length', ('geo:dist', {}), {
+                        'doc': 'The official overall vessel length'}),
+                    ('beam', ('geo:dist', {}), {
+                        'doc': 'The official overall vessel beam'}),
+                    ('flag', ('iso:3166:cc', {}), {
```
Comment (vEpiphyte) [inline synapse/models/transport.py:141]: another place where the existing pol:iso2 type works

File: synapse/models/gov/intl.py (hunk 4 above)
Comment (invisig0th) [reply synapse/models/gov/intl.py:11]: pol:iso2 is a terribly mis-named type that i would like us to move away from.  the first step is to have something else to move to. :)

File: synapse/tests/test_model_geospace.py [hunk 7]
```diff
@@ -428,3 +431,14 @@ async def test_geojson(self):
 
             opts = {'vars': {'geojson': geojson2}}
             nodes = await core.nodes('[ geo:place=* :geojson=$geojson ]', opts=opts)
+
+    async def test_geo_dist_offset(self):
```
Comment (vEpiphyte) [inline synapse/tests/test_model_geospace.py:435]: just roll this into an existing test which loads GeoTstModule

File: synapse/models/geospace.py (hunk 3 above)
Comment (invisig0th) [reply synapse/models/geospace.py:194]: also added valu=text

File: synapse/tests/test_model_geospace.py (hunk 7 above)
Comment (invisig0th) [reply synapse/tests/test_model_geospace.py:435]: No other test method has a name that could logically include this. Unless you're saying we should rename another test to be more generic, i dont see how this is reasonable.

File: synapse/lib/task.py (hunk 5 above)
Comment (invisig0th) [reply synapse/lib/task.py:100]: ( adding a test to demonstrate )

**Pull request merged**